src/Tools/dotnet-openapi/src/Commands/BaseCommand.cs

```diff
@@ -0,0 +1,294 @@
+            var result = new List<Tuple<string, string>>();
+            foreach (var package in packages)
+            {
+                var tmp = package.Split(':', StringSplitOptions.RemoveEmptyEntries);
+                Debug.Assert(tmp.Length == 2);
+                result.Add(new Tuple<string, string>(tmp[0], tmp[1]));
+            }
+
+            return result;
+        }
+
+        private async Task WriteToFile(Stream content, string destinationPath, bool overwrite)
+        {
+            var destinationExists = File.Exists(destinationPath);
+            if (destinationExists && !overwrite)
+            {
+                await Out.WriteAsync($"Not overwriting existing file '{destinationPath}'.");
+                return;
+            }
+
+            await Out.WriteAsync($"Downloading to '{destinationPath}'.");
+            var reachedCopy = false;
+            try
+            {
+                if (destinationExists)
+                {
+                    // Check hashes before using the downloaded information.
+                    var downloadHash = DownloadFileExtensions.GetHash(content);
+
+                    byte[] destinationHash;
+                    using (var destinationStream = File.OpenRead(destinationPath))
+                    {
+                        destinationHash = DownloadFileExtensions.GetHash(destinationStream);
+                    }
+
+                    var sameHashes = downloadHash.Length == destinationHash.Length;
+                    for (var i = 0; sameHashes && i < downloadHash.Length; i++)
+                    {
+                        sameHashes = downloadHash[i] == destinationHash[i];
+                    }
+
+                    if (sameHashes)
+                    {
+                        await Out.WriteAsync($"Not overwriting existing and matching file '{destinationPath}'.");
+                        return;
+                    }
+                }
+                else
+                {
+                    // May need to create directory to hold the file.
+                    var destinationDirectory = Path.GetDirectoryName(destinationPath);
+                    if (!string.IsNullOrEmpty(destinationDirectory))
+                    {
+                        Directory.CreateDirectory(destinationDirectory);
+                    }
+                }
+
+                // Create or overwrite the destination file.
+                reachedCopy = true;
+                using (var outStream = File.Create(destinationPath))
+                {
+                    await content.CopyToAsync(outStream);
+
+                    await outStream.FlushAsync();
+                }
+            }
+            catch (Exception ex)
+            {
+                await Error.WriteAsync($"Downloading failed.");
+                await Error.WriteAsync(ex.ToString());
+                if (reachedCopy)
+                {
+                    File.Delete(destinationPath);
```

Comment:
What about the directory that was created?

Reply:
What if multiple levels of directories were created? What if other files were put into that directory, or some other directory somewhere along the chain we created? Doing "the right thing" in this case is actually kind of complicated, and I don't think it's really worth it since this failure mode is much less likely than getting non-200 from the result. CC @dougbu (who wrote this code for a different thing) for opinions.

Reply:
Yea, it's hard to guarantee that you're leaving the file system the same way you found it. It's fine if this is what we expect, but just wanted to call out the behaviour.